File: RDCPanel/RDCPanel.Web/Modules/ServerTypes/RDTeam/RdTeamsForm.ts
import { StringEditor, ImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface RdTeamsForm {
    Name: StringEditor;
    JobTitle: StringEditor;
    Department: StringEditor;
    Extension: StringEditor;
    Email: StringEditor;
    Image: ImageUploadEditor;
    Branch: StringEditor;
    Initial: StringEditor;
}

export class RdTeamsForm extends PrefixedContext {
    static readonly formKey = 'RDTeam.RdTeams';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!RdTeamsForm.init)  {
            RdTeamsForm.init = true;

            var w0 = StringEditor;
            var w1 = ImageUploadEditor;

            initFormType(RdTeamsForm, [
                'Name', w0,
                'JobTitle', w0,
                'Department', w0,
                'Extension', w0,
                'Email', w0,
                'Image', w1,
                'Branch', w0,
                'Initial', w0
            ]);
        }
    }
}

File: RDCPanel/RDCPanel.Web/Modules/RDTeam/RdTeams/RdTeamsForm.cs
using Serenity.ComponentModel;

namespace RDCPanel.RDTeam.Forms;

[FormScript("RDTeam.RdTeams")]
[BasedOnRow(typeof(RdTeamsRow), CheckNames = true)]
public class RdTeamsForm
{
    public string Name { get; set; }
    public string JobTitle { get; set; }
    public string Department { get; set; }
    public string Extension { get; set; }
    public string Email { get; set; }
    public string Image { get; set; }
    public string Branch { get; set; }
    public string Initial { get; set; }
}